stats/builder.go

```diff
@@ -0,0 +1,110 @@
+package stats
+
+import (
+	"expvar"
+)
+
+// Initializes a new Builder and associates with the specified registry.
+func newBuilder(k string, tags map[string]string, r registryClient) Builder {
+	impl := &expvar.Map{}
+	impl.Init()
+
+	builder := &statistics{
+		registry:   r,
+		key:        k,
+		tags:       tags,
+		impl:       impl,
+		refsCount:  0,
+		intVars:    map[string]*expvar.Int{},
+		stringVars: map[string]*expvar.String{},
+		floatVars:  map[string]*expvar.Float{},
+		types:      map[string]string{},
+	}
+
+	return builder
+}
+
+// Checks whether the receiver has already been built and returns an error if it has
+func (s *statistics) checkNotBuilt() error {
+	if s.built {
+		return ErrAlreadyBuilt
+	} else {
```

Comment:
delete the else, the branch condition cannot reach the else

if true {
   return somethign
}
return not nil